src/securejoin.rs

```diff
@@ -477,24 +499,31 @@ pub(crate) async fn handle_securejoin_handshake(
                 ChatId::create_for_contact(context, contact_id).await?;
             }
             context.emit_event(EventType::ContactsChanged(Some(contact_id)));
-            if let Some(group_chat_id) = group_chat_id {
+            if let Some(joining_chat_id) = joining_chat_id {
                 // Join group.
-                chat::add_contact_to_chat_ex(context, Nosync, group_chat_id, contact_id, true)
+                chat::add_contact_to_chat_ex(context, Nosync, joining_chat_id, contact_id, true)
                     .await?;
-                let is_group = true;
-                inviter_progress(context, contact_id, group_chat_id, is_group)?;
-                // IMAP-delete the message to avoid handling it by another device and adding the
-                // member twice. Another device will know the member's key from Autocrypt-Gossip.
-                Ok(HandshakeMessage::Done)
+                let chat = Chat::load_from_db(context, joining_chat_id).await?;
+
+                inviter_progress(context, contact_id, joining_chat_id, chat.typ)?;
+                if chat.typ == Chattype::OutBroadcast {
+                    // For broadcasts, we don't want to delete the message,
+                    // because the other device should also internally add the member
+                    // and see the key (because it won't see the member via autocrypt-gossip).
```

Comment:
Why can't we send Autocrypt-Gossip for broadcasts? Asking just to understand the reason for this difference with groups (and maybe it makes sense to document it here)

Reply:
The problem with leaving `vg-request-with-auth` on IMAP is the second device going online much later and readding the subscriber that has already left. Do some tests break if we make this like it's done for groups?

OTOH we shouldn't immediately delete the message until we replied to it, otherwise the second device has no chance to finish the protocol.